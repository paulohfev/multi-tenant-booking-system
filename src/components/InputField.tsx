export type InputFieldProps = {
  label: string;
  name: string;
  type: string;
  value: string;
  onChange: (e: React.ChangeEvent<HTMLInputElement>) => void;
  min?: string;
}

const InputField: React.FC<InputFieldProps> = ({
  label,
  name,
  type,
  value,
  onChange,
  min,
}) => {
  return (
    <div>
      <label
        htmlFor={name}
        className="block text-sm font-medium text-gray-700"
      >
        {label}
      </label>
      <input
        className="mt-1 block w-full rounded-md border-gray-300 p-2 shadow-sm focus:border-blue-500 focus:ring-blue-500 text-black"
        type={type}
        id={name}
        name={name}
        value={value}
        onChange={onChange}
        min={min}
        required
      />
    </div>
  )
}

export default InputField;