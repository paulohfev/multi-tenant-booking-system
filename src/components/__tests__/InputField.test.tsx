import { render } from '@testing-library/react'
import InputField from '../InputField'

describe('InputField', () => {
  it('should match snapshot with required props', () => {
    const { container } = render(<InputField label='Test Label' name='test' type='text' value='' onChange={() => {}} />)
    expect(container).toMatchSnapshot()
  })

  it('should match snapshot with min value', () => {
    const { container } = render(
      <InputField label='Test Label' name='test' type='date' value='' onChange={() => {}} min='2024-03-21' />
    )
    expect(container).toMatchSnapshot()
  })
})
