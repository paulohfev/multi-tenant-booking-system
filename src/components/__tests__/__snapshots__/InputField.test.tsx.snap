// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputField should match snapshot with min value 1`] = `
<div>
  <div>
    <label
      class="block text-sm font-medium text-gray-700"
      for="test"
    >
      Test Label
    </label>
    <input
      class="mt-1 block w-full rounded-md border-gray-300 p-2 shadow-sm focus:border-blue-500 focus:ring-blue-500 text-black"
      id="test"
      min="2024-03-21"
      name="test"
      required=""
      type="date"
      value=""
    />
  </div>
</div>
`;

exports[`InputField should match snapshot with required props 1`] = `
<div>
  <div>
    <label
      class="block text-sm font-medium text-gray-700"
      for="test"
    >
      Test Label
    </label>
    <input
      class="mt-1 block w-full rounded-md border-gray-300 p-2 shadow-sm focus:border-blue-500 focus:ring-blue-500 text-black"
      id="test"
      name="test"
      required=""
      type="text"
      value=""
    />
  </div>
</div>
`;
